sdk/lib/atl/atlwin.h

```diff
@@ -320,9 +320,32 @@ class CWindow
         int wndCenterHeight = wndCenterRect.bottom - wndCenterRect.top;
         int wndWidth = wndRect.right - wndRect.left;
         int wndHeight = wndRect.bottom - wndRect.top;
+        int xPos = wndCenterRect.left + ((wndCenterWidth - wndWidth + 1) >> 1);
+        int yPos = wndCenterRect.top + ((wndCenterHeight - wndHeight + 1) >> 1);
+        RECT rcWork;
+        SystemParametersInfo(SPI_GETWORKAREA, 0, &rcWork, 0);
+
+        if (xPos + wndWidth > rcWork.right)
+        {
+            xPos = rcWork.right - wndWidth;
+        }
+        else if (xPos < 0)
```

Comment:
Shouldn't we use the left part of the work area here instead as the boundary? (instead of 0).
+ Same question for the y coordinate.

Reply:
probably, yes.